bpdm-gate/src/main/kotlin/org/eclipse/tractusx/bpdm/gate/entity/AlternativePostalAddress.kt

```diff
@@ -72,4 +72,34 @@ class AlternativePostalAddress(
      */
     @Column(name = "alt_delivery_service_number")
     val deliveryServiceNumber: String?
-)
+) {
+    override fun equals(other: Any?): Boolean {
```

Comment:
It's a neat idea but I'm not sure if we want to override the default equals method for this. Here, we define our own equals in a specific context. Like, it is not really completely equal since technical fields like timestamps and identifier could be still different. In other contexts we might want to treat the meaning of equality differently. I would propose to move to another service which contains the equality methods.

I'm open for discussion though, would like to see the input of the other reviewers

Reply:
Firstly I was creating different methods which would compare the upserted object with the newly inserted object field by field, but thought that wasn't the right and cleanest way to do it. Asked for some tips from Fabio how could I do it a different way, and decided with this idea. Probably there is some other clean and easiest way to do such thing in Kotlin, but didn't found out. Any tips are welcome :)

Reply:
If I'm not mistaken you're considering *all* properties for equality check.
There is no ID or timestamps because this is an Embeddable, not an Entity.
I would even make `AlternativePostalAddress` a data class like `GeographicCoordinate`, then we can get rid of manual `equals` and `hashCode` implementations.